File: 프로그래머스/0/120850. 문자열 정렬하기 （1）/문자열 정렬하기 （1）.py
def solution(my_string):
    # 숫자를 저장할 리스트 초기화
    answer = []

    # 문자열을 한 글자씩 순회
    for m in my_string:
        # 현재 문자가 숫자인지 확인 (isnumeric() 사용)
        if m.isnumeric():
            # 숫자인 경우 정수형(int)으로 변환하여 리스트에 추가
            answer.append(int(m))

    # 숫자 리스트를 오름차순 정렬
    answer.sort()

    # 정렬된 숫자 리스트 반환
    return answer

# ✅ 예제 테스트 실행
print(solution("hi12392"))
# 결과: [1, 2, 2, 3, 9] (숫자만 추출 후 정렬)

print(solution("p2o4i8gj2"))
# 결과: [2, 2, 4, 8] (숫자만 추출 후 정렬)

print(solution("abcde0"))
# 결과: [0] (숫자 '0'만 존재)

File: 프로그래머스/0/120850. 문자열 정렬하기 （1）/문자열 정렬하기 （1）.java
import java.util.*;

public class Solution {

    public static int[] solution(String my_string) {
        List<Integer> numbers = new ArrayList<>();

        // 문자열을 한 글자씩 확인
        for (int i = 0; i < my_string.length(); i++) {
            char c = my_string.charAt(i);
            if (Character.isDigit(c)) {
                numbers.add(c - '0');  // 문자 숫자를 정수로 변환
            }
        }

        // 오름차순 정렬
        Collections.sort(numbers);

        // List<Integer> → int[] 변환
        int[] result = new int[numbers.size()];
        for (int i = 0; i < numbers.size(); i++) {
            result[i] = numbers.get(i);
        }

        return result;
    }

    // ✅ 테스트 실행
    public static void main(String[] args) {
        System.out.println(Arrays.toString(solution("hi12392")));   // [1, 2, 2, 3, 9]
        System.out.println(Arrays.toString(solution("p2o4i8gj2"))); // [2, 2, 4, 8]
        System.out.println(Arrays.toString(solution("abcde0")));    // [0]
    }
}
